hoomd/snapshot.py

```diff
@@ -109,3 +109,65 @@ def replicate(self, nx, ny, nz):
 
     def _broadcast_box(self):
         self._cpp_obj._broadcast_box(self._comm.cpp_mpi_conf)
+
+    @classmethod
+    def _from_gsd_snapshot(cls, gsd_snap):
+        gsd_snap.validate()
+        if self._device.communicator.rank == 0:
```

Comment:
There is no self for a classmethod

Reply:
OK, I must've misunderstood @joaander's suggestion:
>The whole set routine needs to be wrapped in an: `if self._device.communicator.rank == 0:` block.

I'm not sure what was the correct way to add this.